opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -369,7 +369,10 @@ namespace Opm {
 
                 const auto& events = schedule()[reportStepIdx].wellgroup_events();
                 const bool event = report_step_starts_ && events.hasEvent(well->name(), effective_events_mask);
-                if (event) {
+                const bool dyn_status_change = this->wellState().well(well->name()).status
```

Comment:
The `dyn_status_change` variable was a real mouthful - but I have no alternative suggestion.